interpreter/activation.go

```diff
@@ -108,15 +97,12 @@ func (a *mapActivation) ResolveName(name string) (ref.Val, bool) {
 		// Resolve a lazily bound value.
 		case func() ref.Val:
 			val := object.(func() ref.Val)()
+			val = a.adapter.NativeToValue(val)
```

Comment:
Will `a.adapter` be non-nil all the time?

Reply:
Yes, it should be, but I'll add an input test to make sure it's non-nil.